ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -101,6 +101,17 @@ export function WorkflowLogsViewer({workflow, nodeId, initialPodName, container,
     useEffect(() => {
         setError(null);
         setLoaded(false);
+
+        if (archived) {
+            // replace nodeId with a nodeId which is associated with the podName, so we can also switch bewteen archived logs if a differnet node was selected.
+            nodeId = podNamesToNodeIDs.get(podName);
+
+            // if a podName is provided but no valid nodeId is found, clear the podName. So the selector is automatically set to 'All'
```

Comment:
```suggestion
            // if a podName is provided but has no valid nodeId, clear the podName so that the selector is automatically set to 'All'
```
- clarify